qiita_db/base.py

```diff
@@ -128,28 +130,36 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
-
-    def __init__(self, id_):
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        # The value that we want is the result of the last SQL query,
+        # and it is stored in the first value of the first row
+        return trans.execute()[-1][0][0]
```

Comment:
Please read the comment. This is exactly what @wasade commented before and I added a comment.

I have to access a 3 nested list in some way, and those indexes are not provided by anybody, they just happen to be those:
 - -1 is the SQL query (last one)
 - 0 which row on that query I'm searching the result
 - 0 which value on that row is the result that I want

If you don't like this solution, I'm open to ideas. Applies to all.